cpp/src/io/csv/csv_writer.cu

```diff
@@ -31,26 +31,31 @@
 // This is called by the write_csv method below.
 //
 // Parameters:
-// - column:    The column to be converted.
+// - column: The column to be converted.
+// - row_offset: Number entries to from the beginning to skip; must be multiple of 8.
+// - rows:  How many rows from the offset that should be converted for this column.
 // - delimiter: Separator to append to the column strings
 // - null_representation: String to use for null entries
 // - true_string: String to use for 'true' values in boolean columns
 // - false_string: String to use for 'false' values in boolean columns
 // Return: NVStrings instance formated for CSV column output.
 //
-NVStrings* column_to_strings_csv(const gdf_column* column, const char* delimiter, const char* null_representation, const char* true_string, const char* false_string )
+NVStrings* column_to_strings_csv(const gdf_column* column, gdf_size_type row_offset, gdf_size_type rows,
+                                 const char* delimiter, const char* null_representation,
+                                 const char* true_string, const char* false_string )
 {
     NVStrings* rtn = nullptr;
-    gdf_size_type rows = column->size;
     gdf_valid_type* valid = column->valid;
+    if( valid )
+        valid += ((row_offset+7)/8);
```

Comment:
Use `cudf::util::detail::bit_container_index<gdf_valid_type, gdf_index_type>(row_offset)` from `bit_util.cuh`.

If you must do the math yourself, the approach you use here can overflow, so use `cudf::util::div_rounding_up_safe()` to avoid overflow. https://github.com/rapidsai/cudf/blob/fa94941c8b61927a6fea7880a9dc322c986b9a20/cpp/src/utilities/integer_utils.hpp#L98

Also don't hard-code `8` -- use `GDF_VALID_BITSIZE`.

Reply:
Thanks for this Mark. Searching for usage, I found this alternative one `packed_bit_sequence_size_in_bytes()` here: 
https://github.com/rapidsai/cudf/blob/fa94941c8b61927a6fea7880a9dc322c986b9a20/cpp/src/utilities/bit_util.cuh#L114
This is closer to what I want.

Reply:
That function is unused and undocumented. I didn't know it existed, and I would say since it has never been used it shouldn't be there. Also, the name is awkward. 

What is it you are trying to achieve, exactly? The intent is not clear.

Reply:
Pointing to the next chunk of valid bytes.